src/geometry/euclidean/oriented_angle.lean

```diff
@@ -788,6 +790,54 @@ begin
   rw [←submonoid.coe_mul, mul_left_inv, submonoid.coe_one, mul_one]
 end
 
+/-- The inner product of two vectors is the product of the norms and the cosine of the oriented
+angle between the vectors. -/
+lemma inner_eq_norm_mul_norm_mul_cos_oangle (x y : V) :
+  ⟪x, y⟫ = ∥x∥ * ∥y∥ * real.angle.cos (hb.oangle x y) :=
+begin
+  by_cases hx : x = 0,
+  { simp [hx] },
+  { by_cases hy : y = 0,
```

Comment:
The [style guide](https://leanprover-community.github.io/contribute/style.html) has an exception for situations like this to avoid indentation:
> "Very short goals can be closed right away using swap or work_on_goal if needed, to avoid additional indentation in the rest of the proof."

```suggestion
  by_cases hx : x = 0, { simp [hx] },
  by_cases hy : y = 0, { simp [hy] },
  -- and then unindent the remainder
```

Reply:
Done, also with the side goal from rewriting with `complex.cos_arg` that's closed by `{ simp [hx, hy] }`.